### ISSUE TITLE
Android 13 MapIsReady().

### ISSUE THREAD

**Issue by david-hov** [liodali/osm_flutter#433]
ANDROID 13 , MUI GLOBAL version 14.0.1
Sometimes MapIsReady returns false, and not ending Loading
OSMFlutter(
        controller: mapController,
        onMapIsReady: (isReady) {
          if (isReady) {
            print('ready');
          }
        },
        mapIsLoading: Container(
            decoration: BoxDecoration(
          image: DecorationImage(
            image: AssetImage('assets/screenBackground.png'),
            fit: BoxFit.cover,
            colorFilter: ColorFilter.mode(
              Colors.grey.withOpacity(0.5),
              BlendMode.dstIn,
            ),
          ),
        )),
        osmOption: OSMOption(
          showContributorBadgeForOSM: true,
          zoomOption: ZoomOption(initZoom: 12),
          userLocationMarker: UserLocationMaker(
            personMarker: MarkerIcon(
              icon: Icon(
                Icons.directions_bus,
                color: Colors.red,
                size: 65.sp,
              ),
            ),
            directionArrowMarker: MarkerIcon(
              icon: Icon(
                Icons.navigation,
                size: 65.sp,
                color: Colors.orange,
              ),
            ),
          ),
          showZoomController: true,
          markerOption: MarkerOption(
              defaultMarker: MarkerIcon(
            icon: Icon(
              Icons.person_pin_circle,
              color: Colors.blue,
              size: 65.sp,
            ),
          )),
        ));

**Comment by liodali** [comment]
do you have any logs error?

**Comment by david-hov** [comment]
no errors

**Comment by david-hov** [comment]
E/osm     (25118): osm flutter plugin create
E/osm     (25118): osm flutter plugin start
E/osm     (25118): osm flutter plugin resume
I/PlatformViewsController(25118): Hosting view in view hierarchy for platform view: 0
E/FrameEvents(25118): updateAcquireFence: Did not find frame.
W/Parcel  (25118): Expecting binder but got null!
E/FileUtils(25118): err write to mi_exception_log

**Comment by liodali** [comment]
that's wierd I have the same log and map always shows 
if you can provide your all page to see maybe your rebuiling the map during initialisation of the map

**Comment by david-hov** [comment]
> that's wierd I have the same log and map always shows if you can provide your all page to see maybe your rebuiling the map during initialisation of the map

no it's not rebuilding, as I put OSM map in home start screen, and no have states to rebuild

**Comment by david-hov** [comment]
> that's wierd I have the same log and map always shows if you can provide your all page to see maybe your rebuiling the map during initialisation of the map

I/OsmDroid( 9961): Using tile source: Mapnik
I/OsmDroid( 9961): Tile cache increased from 0 to 9
E/osm     ( 9961): osm flutter plugin create
E/osm     ( 9961): osm flutter plugin start
E/osm     ( 9961): osm flutter plugin resume
I/PlatformViewsController( 9961): Hosting view in view hierarchy for platform view: 0
E/FrameEvents( 9961): updateAcquireFence: Did not find frame.
E/FileUtils( 9961): err write to mi_exception_log
W/Parcel  ( 9961): Expecting binder but got null!

it worked before last version of OSM flutter, after updateing to the latest, this happened

**Comment by liodali** [comment]
i didnt change anything in android side 
except fixing some bugs 
try open android project and debug it maybe you can help to find issue or we can have private session together (video call)  to see the issue

**Comment by david-hov** [comment]
> i didnt change anything in android side except fixing some bugs try open android project and debug it maybe you can help to find issue or we can have private session together (video call) to see the issue

Device is not emuliator I just want to run on my XIaomi, and then I saw the issue. On other devices it seems working.

**Comment by Legendorik** [comment]
Yes, I can confirm the bug. Updated from version 0.44 to 0.60.1 and now the map is permanently stuck in a loading state.

OnMapIsReady callback is triggered twice in my case (is it supposed to?). First time with result = false and second time with result = true. If you use setState inside OnMapIsReady callback or otherwise rebuild the widget with map, map will be stuck in a loading state. Without rebuilds everything works fine. I think this behavior should not happen in any case. In earlier versions everything was okay.

The issue can be reproduced on android emulator 
![изображение](https://github.com/liodali/osm_flutter/assets/42907177/d72a5301-5e19-4cd6-95f9-1d411966000d)

**Comment by liodali** [comment]
I think  mapIsReady is not good solution 
I will think in new api or maybe improve that i will be called only first time when map initialized

**Comment by Legendorik** [comment]
Out of curiosity: what is the limitation of android? 
I tried removing the following code
https://github.com/liodali/osm_flutter/blob/f63fcaa8f67c986c417182a540899a8d46021923/lib/src/osm_flutter.dart#L147-L150C6
And it seems to work that way.

I got the impression from the commit names that this code is needed to support hot reloading, but we have an option `androidHotReloadSupport` and it's not used here for some reason.

**Comment by liodali** [comment]
you can activated but I don't recommand it
In general if you work with ValueNotifier you will not face any issue

* For limitation that osmdroid use android context to show the map, when widget rebuild, the map loose the context
for unknown reason.

* `androidHotReloadSupport`  used  in internal widget  <<mobile_osm_flutter>>

* another advice that extract the map from you main widget into sepaerated statelessWidget and pass to it controller 
thats will prevent of reloading the map and keep the other UI part in main widget 

as I test it now, maybe there some changes happened to platformView so 
I will remove that piece of code in next version

**Comment by liodali** [comment]
we published new version to fix that : `0.60.2`

**Comment by Legendorik** [comment]
Everything's fine now.
